### PR TITLE
Fixes #19908: Strict mode of file_key_value_present_option does not introduce unwanted escpaing characters

### PR DESCRIPTION
**Pull request by Fdall**
https://issues.rudder.io/issues/19908

### CODE REVIEW

File: tree/30_generic_methods/file_key_value_present_option.cf [hunk 1]
```diff
@@ -39,7 +39,7 @@ bundle agent file_key_value_present_option(file, key, value, separator, option)
       "args"              slist => { "${file}", "${key}", "${value}","${separator}", "${option}" };
 
       "report_param"      string => join("_", args);
-      "full_class_prefix" string => canonify("file_key_value_present_${report_param}");
+      "full_class_prefix" string => canonify("file_key_value_present_option_${report_param}");
```
Comment (ncharles) [inline tree/30_generic_methods/file_key_value_present_option.cf:42]: You cannot change this: the advertised class_prefix is file_key_value_present_

**Review by ncharles — CHANGES_REQUESTED** [review]
you shouldn't change the class_prefix

**Comment by Fdall** [comment]
PR updated with a new commit

File: tree/30_generic_methods/file_key_value_present_option.cf [hunk 2]
```diff
@@ -42,6 +42,9 @@ bundle agent file_key_value_present_option(file, key, value, separator, option)
       "full_class_prefix" string => canonify("file_key_value_present_${report_param}");
       "class_prefix"      string => string_head("${full_class_prefix}", "1000");
 
+      "updated_full_class_prefix" string => canonify("file_key_value_present_optin_${report_param}");
```
Comment (Fdall) [inline tree/30_generic_methods/file_key_value_present_option.cf:45]:
```suggestion
      "updated_full_class_prefix" string => canonify("file_key_value_present_option_${report_param}");
```

File: tree/30_generic_methods/file_key_value_present_option.cf [hunk 3]
```diff
@@ -42,6 +42,9 @@ bundle agent file_key_value_present_option(file, key, value, separator, option)
       "full_class_prefix" string => canonify("file_key_value_present_${report_param}");
       "class_prefix"      string => string_head("${full_class_prefix}", "1000");
 
+      "updated_full_class_prefix" string => canonify("file_key_value_present_option_${report_param}");
+      "updated_class_prefix"      string => string_head("${full_class_prefix}", "1000");
```
Comment (Fdall) [inline tree/30_generic_methods/file_key_value_present_option.cf:46]:
```suggestion
      "updated_class_prefix"      string => string_head("${updated_full_class_prefix}", "1000");
```

**Review by ncharles — APPROVED** [review]

**Comment by Fdall** [comment]
PR rebased

**Comment by Fdall** [comment]
PR rebased

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
OK, squash merging this PR

**Comment by Fdall** [comment]
OK, merging this PR

**Pull request merged**